File: c#/SingledListList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SingleListCS
{
    public class Node
    {
        public object Data { set; get; }
        public Node Next {set; get; }

        public Node(object data)
        {
            Data = data;
            Next = null;
        }
    }

    public class LinkedList
    {
        Node Head { set; get; }
        Node Tail { set; get; }


        public LinkedList()
        {
            Head = null;
            Tail = null;
        }

        public void AddToTail(Node data)
        {
            if (Head == null)
            {
                Head = data;
                Tail = data;
            }
            else
            {
                Tail.Next = data;
                Tail = data;
            }
        }

        public void removeLastNode()
        {
            Node prevNode = null;
            Node curNode = Head;
            while (curNode != Tail)
            {
                prevNode = curNode;
                curNode = curNode.Next;
            }

            Tail = prevNode;
            prevNode.Next = null;
        }

        public void Traverse()
        {
            Node curNode = Head;
            while (curNode != null)
            {
                Console.WriteLine(curNode.Data);
                curNode = curNode.Next;
            }
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            LinkedList list = new LinkedList();
            list.AddToTail(new Node("abc"));
            list.AddToTail(new Node("def"));
            list.Traverse();
            list.removeLastNode();
            Console.WriteLine("===");
            list.Traverse();
            Console.Read();
        }
    }
}

File: python/SingledListList.py
class Node :
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None
        self.tail = None
    
    def addToTail(self, data):
        node = Node(data)
        if self.head == None :
            self.head = node
            self.tail = node
        else:
            self.tail.next = node
            self.tail = node
            
    def traverse(self):
        curNode = self.head;
        while curNode is not None:
            print curNode.data
            curNode = curNode.next
            
def say_hello():
    print 'Hello, World'

for i in xrange(5):
    say_hello()

print "\n"    
list = LinkedList()

list.addToTail("abcd")
list.addToTail("efgh")
list.traverse()

